src/main/java/com/commercetools/sync/categories/impl/CategorySyncImpl.java

```diff
@@ -0,0 +1,129 @@
+                if (oldCategory == null) {
+                    currentSyncResult = createCategory(newCategoryDraft);
+                } else {
+                    currentSyncResult = CategorySyncUtils.buildActions(oldCategory, newCategoryDraft, typeService);
+                    final List<UpdateAction<Category>> updateActions = currentSyncResult.getUpdateActions();
+                    if (!updateActions.isEmpty()) {
+                        updateCategory(oldCategory, currentSyncResult);
+                    }
+                }
+            }
+            syncResult.mergeStatistics(currentSyncResult);
+        }
+        LOGGER.info(getSummary(syncResult));
+    }
+
+    @Nullable
+    private SyncResult<Category> createCategory(@Nonnull final CategoryDraft newCategory) {
+        SyncResult<Category> syncResult = SyncResult.emptyResult();
+        try {
+            categoryService.createCategory(newCategory);
+            createdCategories++;
+        } catch (Exception e) {
+            syncResult = SyncResult.ofError(SyncError.of(null, format("Failed to create category with external id" +
+                                    " '%s' in CTP project with key '%s",
+                            newCategory.getExternalId(), options.getCtpProjectKey()), e));
+            failedCategories++;
+        }
+        return syncResult;
+    }
+
+    @Nullable
+    private SyncResult<Category> updateCategory(@Nonnull final Category category,
+                                                @Nonnull final SyncResult<Category> syncResult) {
+        try {
+            categoryService.updateCategory(category, syncResult.getUpdateActions());
+            updatedCategories++;
+        } catch (Exception e) {
+            syncResult.addError(
+                    SyncError.of(category.getId(), format("Failed to update category with id" +
+                                    " '%s' in CTP project with key '%s",
+                            category.getId(), options.getCtpProjectKey()), e));
+            failedCategories++;
+        }
+        return syncResult;
+    }
+
+
+    @Override
+    public String getSummary(@Nonnull final SyncResult<Category> syncResult) {
+        return format("Category Sync completed successfully!\n" +
+                        "Sync Summary:\n%s(%s) categories were processed in total." +
+                        "\n\t+ (%s) categories created." +
+                        "\n\t+ (%s) categories updated." +
+                        "\n\t+ (%s) categories failed to sync." +
+                        "\n\n" +
+                        "Detailed Report of Sync Statistics:\n" +
+                        "%s\n%s",
+                getHorizontalLine(15),
+                processedCategories,
+                createdCategories,
+                updatedCategories,
+                failedCategories,
+                getHorizontalLine(35),
+                syncResult.getResultStatistics());
```

Comment:
Is there a specific structure for this JSON statistics object or how it should look like? e.g. on the nodejs sync? /cc @lojzatran